<?xml version="1.0"?>
<launch>
    <arg name="vocabulary_path" default="$(find orb_slam2_cuda)/Vocabulary/ORBvoc.txt" />
    <arg name="camera_setting_path" default="$(find my_pipeline)/calibration/s20_wide_openvslam.yaml" />
    <arg name="bUseViewer" default="true" />
    <arg name="bEnablePublishROSTopic" default="true" />


    <arg name="cam_name" value="s20cam_wide" />

    <!-- Start the GSCAM node -->
    <env name="GSCAM_CONFIG" value="rtspsrc location=rtsp://192.168.1.99:5554/playlist.m3u latency=100  ! typefind ! rtph264depay ! h264parse ! avdec_h264 ! videoconvert" />
    <node pkg="gscam" type="gscam" name="$(arg cam_name)">
         <param name="camera_name" value="$(arg cam_name)" />
         <param name="camera_info_url" value="package://my_pipeline/calibration/ost_wide.yaml" />
         <param name="frame_id" value="$(arg cam_name)" />
         <remap from="/camera/image_raw" to="/$(arg cam_name)/image_raw" />
    </node>

    <node pkg="image_proc" type="image_proc" name="creative_image_proc" ns="$(arg cam_name)" />

    <arg name="node_start_delay" default="20" />

    <node name="orb_slam2_cuda" pkg="orb_slam2_cuda" type="Mono" output="screen"
    	<param name="voc_file" value="$(arg vocabulary_path)"/>
    	<param name="settings_file" value="$(arg camera_setting_path)"/>
    	<remap from="/camera/image_raw" to="/$(arg cam_name)/image_rect"/>
  	</node>

</launch>
